storages/jsonl-storage/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+                    .map_storage_err(JsonlStorageError::FileNotFound.to_string())?
+                    .to_owned();
+
+                self.fetch_schema(table_name.as_str())?
+                    .map_storage_err(JsonlStorageError::TableDoesNotExist.to_string())
+            })
+            .collect::<Result<Vec<Schema>>>()
+    }
+
+    async fn fetch_data(&self, table_name: &str, target: &Key) -> Result<Option<DataRow>> {
+        let row = self.scan_data(table_name)?.find_map(|result| {
+            result
+                .map(|(key, row)| (&key == target).then_some(row))
+                .unwrap_or(None)
+        });
+
+        Ok(row)
+    }
+
+    async fn scan_data(&self, table_name: &str) -> Result<RowIter> {
+        self.scan_data(table_name)
+    }
+}
+
+fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
+where
+    P: AsRef<Path>,
+{
+    let file = File::open(filename)?;
+    Ok(io::BufReader::new(file).lines())
+}
+
+#[async_trait(?Send)]
+impl StoreMut for JsonlStorage {
+    async fn insert_schema(&mut self, schema: &Schema) -> Result<()> {
+        let data_path = self.data_path(schema.table_name.as_str());
+        File::create(data_path).map_storage_err()?;
+
+        if schema.column_defs.is_some() {
+            let schema_path = self.schema_path(schema.table_name.as_str());
+            let ddl = schema.to_ddl();
+            let mut file = File::create(schema_path).map_storage_err()?;
+            write!(file, "{ddl}").map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn delete_schema(&mut self, table_name: &str) -> Result<()> {
+        let data_path = self.data_path(table_name);
+        if data_path.exists() {
+            remove_file(data_path).map_storage_err()?;
+        }
+
+        let schema_path = self.schema_path(table_name);
+        if schema_path.exists() {
+            remove_file(schema_path).map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn append_data(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        #[derive(Iterator)]
+        enum JsonIter<I1, I2> {
+            Map(I1),
+            Vec(I2),
+        }
+
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist.to_string())?;
+        let table_path = JsonlStorage::data_path(self, table_name);
+
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .open(table_path)
+            .map_storage_err()?;
+
+        let labels = schema
+            .column_defs
+            .unwrap_or_default()
+            .into_iter()
+            .map(|column_def| column_def.name);
+
+        for row in rows {
+            let json_string = match row {
+                DataRow::Map(hash_map) => JsonIter::Map(hash_map.into_iter()),
+                DataRow::Vec(values) => JsonIter::Vec(labels.clone().zip(values.into_iter())),
```

Comment:
I tried with `&label.zip()` here but looks like `iter_enum` does not work with reference

Reply:
Rather than using `iter_enum`, here because we don't need to return value so how about simply using match to serialize data as jsonvalue independently?
We can deal with `DataRow::Map` and `DataRow::Vec` separately.